data_structures/heap/max_heap.c

```diff
@@ -1,5 +1,6 @@
-#include <stdio.h>
-#include <stdlib.h>
+#include <limits.h>  // to define sizes of integral types
+#include <stdio.h>   // for file related input/output functions
+#include <stdlib.h>  // for memory allocation/freeing functions
```

Comment:
What functions do these headers/libraries include that this algorithm/file uses? Document them here.
For example, the `assert.h` library/header provides the `assert` function. In that case, it should be like:
```c
#include <assert.h>   /// for assert
```